Definicion: (**matriz positiva, negativa y no-negativa definida**):
Se dice que una matriz cuadrada $A$ es positiva definida si todos sus autovalores son positivos, es negativa definida si todos sus autovalores son negativos y es no-negativa definida si todos los autovalores son positivos o 0. Es comun en la literatura encontrar que la definicion de positiva definida es:
Una matriz cuadrada $A$ es positiva definida si para todo $x \in \mathbb{C}^n$, $x^T A x > 0$. Es negativa definida si $\forall x \in \mathbb{C}^n$, $x^T A x < 0$, y  es no negativa definidas si $\forall x \in \mathbb{C}^n$, $x^T A x \ge 0$.
Esta segunda forma de definir si una matriz es positiva (negativa, no-negativa)definida se puede probar que es equivalente a la primera forma. Se deja de tarea al estudiante.

Uno de los teoremas mas poderosos del algebra lineal es el de SVD: Singular Value Decomposition. (descomposicion en valores singulares)

Teorema **SVD**:  Sea $A \in \mathbb{C}^{m \times n}$, entonces existe una matriz $U \in \mathbb{C}^{m \times m}$ ortogonal y una matriz $V \in \mathbb{C}^{n \times n}$ ortogonal y una matriz no-negativa diagonal con valores positivos (reales), $\Sigma \in \mathbb{C}^{m \times n}$ tal que


$$ A = U \Sigma V^*.$$


**Prueba:** (esta prueba se basa en el trabajo de Cornelius Lanczos (hungaro).  Asumamos que $A \in \mathbb{C}^{m \times n}$, podemos construir una matriz por bloques

\begin{eqnarray}
S = \begin{pmatrix}
0  & | & A \\
-- & & --  \\
A^* & | &  0
\end{pmatrix}
\end{eqnarray}
La matriz $S$ es autoadjunta. Es de orden $(m  +n) \times (m+n)$
Entonces por el teorema espectral de matrices autoadjuntas, la matriz tiene $m+n$ autovalores reales y los autovectores son ortogonales. Sean los autovalores $\lambda_i$ y los autovectores $w_i$, $i=1,2, \cdots, m+n$.


Es decir:

$$ S w_i = \lambda_i w_i.$$
Partamos el vector $w_i$ en dos vectores $u_i \in \mathbb{C}^m$ y $v_i \in \mathbb{C}^n$, es decir


\begin{eqnarray}
w_i =
\begin{pmatrix}
u_i  \\ v_i
\end{pmatrix}
\end{eqnarray}
Segun eso, de $Sw_i = \lambda_i w_i$ entontramos

\begin{eqnarray}
A^* u_i = \lambda_i v_i \tag{1A} \\
A v_i = \lambda_i u_i \tag{1B} \\
\end{eqnarray}





Multipliquemos la primera ecuacion por $A$ y la segunda por $A^*$ (a izquierda)

\begin{eqnarray}
A A^* u_i &=& \lambda_i A v_i = \lambda_i \lambda_i v_i = \lambda_i^2 u_i \\
A^* A v_i &=& A^* (\lambda_i u_i) = \lambda_i A^* u_i = \lambda_i^2 v_i
\end{eqnarray}
De aca se observa que $u_i$ es autovector de $A A^*$ y que $v_i$ es autovector de $A^* A$ y que $\lambda_i$ es un autovalor compartido por ambos.

La matriz $A A^* \in \mathbb{C}^{m \times m}$ y la matriz $A^* A \in \mathbb{C}^{n \times n}$.  Tenemos los autovectores $u_i, v_i$. Distingamos la matriz $U_r$ como la matriz de autovectores $u_i$ (columnas) tales que $\lambda_i \ne 0$. Asi mismo llamamos
$U_0$ la matriz de autovectores $u_i$ donde $\lambda_i =0$
Igualmente, distingimos dos matrices. $V_r$ como las columnas $v_i$ tales que $\lambda_i \ne 0$ y $V_0$ como las columnas $v_i$ donde $\lambda_i = 0$
Llamemos a $\Lambda_r$ como la matriz $r \times r$ diagonal con todos los autovalores $|\lambda_r|=\lambda_r$ que no sean 0.
(explico mas adelante por que puedo decir que son positivos)

Entonces podemos formar las siguientes ecuaciones de 1A, 1B


\begin{eqnarray}
A^* U_r = V_r \Lambda_r \quad , \quad A^* U_0 = 0 \\
A V_r = U_r \Lambda_r \quad , \quad A V_0 = 0.
\end{eqnarray}

Definimos la matriz $\Sigma$

\begin{eqnarray}
\Sigma =
\begin{pmatrix}
\Lambda_r & | & 0   \\
-- & &  --  \\
0 & | &  0
\end{pmatrix}
\end{eqnarray}

Vamos a ver que de estas ecuaciones tenemos $A V = U \Sigma \tag{2}$.

Veamos:

$$AV  = A [ V_r | V_0] =[A V_r | A V_0] =[U_r \Lambda_r | 0] =[U_r \Lambda_r + U_0 \times 0 | U_r \times 0 + U_0 \times 0 ]= U \Sigma  .$$






En la ecuacion (2) multiplicamos a derecha por $V^*$, recuerde que $V V^*=I_{n \times n}$ (por que $V$ es ortogonal) Entonces

$$A V V^* = U \Sigma V^*$$
$$A  = U \Sigma V^*$$

En forma de bloques podemos escribir

\begin{eqnarray}
A = ( U_r | U_0)
\begin{pmatrix}
\Lambda_r & | & 0   \\
-- & &  --  \\
0 & | &  0
\end{pmatrix}
\begin{pmatrix}
V_r^* \\ V_0^*
\end{pmatrix}
\end{eqnarray}

Por que podemos decir que $\lambda_i > 0$?
Los $\lambda_i = \sqrt{\lambda_i^2} $ y los vamos a llamar
$\sigma_i = \sqrt{\lambda_i^2}$ se llaman valores singulares.


Como se descompone una matriz $A$ en SVD.

Ejemplo.


In [1]:
import numpy as np
A = np.array([[1, 0, 3], [1,2,4]])
print(A)

[[1 0 3]
 [1 2 4]]


In [4]:
from numpy import linalg
linalg.svd(A)

(array([[-0.552431  , -0.83355863],
        [-0.83355863,  0.552431  ]]),
 array([5.44202115, 1.17660775]),
 array([[-0.25468288, -0.30634156, -0.91721943],
        [-0.23893064,  0.93902322, -0.2472803 ],
        [-0.93704257, -0.15617376,  0.31234752]]))

Ejemplo: A mano (ayudado por Wolfram Alpha o Python).
Sea

\begin{eqnarray}
A =
\begin{pmatrix}
1 & 0 & 3 \\ 1 & 2 & 4
\end{pmatrix}
\end{eqnarray}

Encuentre el SVD. La prueba del teorema nos ayuda a construir el algoritmo para hallar el SVD de $A$, como sigue:

* Construya $A^* A$ y $A A^*$
* Halle los autovalores de la mas pequena de las dos matrices anteriores. Por ejemplo $A^* A$ es $3 \times 3$ y $A A^*$ es $2 \times 2$, entonces halle los autovalores de $A A^*$. Por que el polinomio caracteristico de esta es cuadratico y el otro es cubico. Uno de los autovalores va a ser 0. Las raices cuadradas de estos son los $\sigma_i$.

* Encuentre los autovectores de $A A^*$ estos son los $u_i$ y los de $A^* A$ estos son los $v_i$ que construyen las matrices $U$, $V$.

Comencemos:



* Construyamos $A A^*$.

\begin{eqnarray}
A A^* =
\begin{pmatrix}
1 & 0 & 3 \\ 1 & 2 & 4
\end{pmatrix}
\begin{pmatrix}
1 & 1 \\ 0 & 2 \\ 3 & 4
\end{pmatrix} =
\begin{pmatrix}
10 & 13 \\
13 & 21
\end{pmatrix}
\end{eqnarray}

Nos podemos ayudar de WoframAlpha. Ya les puse una tarea de hallar autovalores/vectores y esto se supone que lo sabemos hacer.

Hagamos la aproximacion con 2 cifras decimales.


$$ \lambda_1^2 = \frac12 ( 31 + \sqrt{797}) = 29.62  .$$
$$ \lambda_2^2 = \frac12 ( 31 - \sqrt{797}) =1.38.$$

Los autovectores (normalizados)

\begin{eqnarray}
u_1 = \begin{pmatrix} 0.55 \\ 0.83 \end{pmatrix} \quad , \quad
u_2 = \begin{pmatrix} -0,83 \\ 0.55 \end{pmatrix}
\end{eqnarray}


* Construyamos ahora $A^* A$.


\begin{eqnarray}
A^* A =
\begin{pmatrix} 1 & 1 \\ 0 & 2 \\ 3 & 4  \end{pmatrix}
\begin{pmatrix} 1 & 0 & 3 \\ 1 & 2 & 4 \end{pmatrix}
=
\begin{pmatrix} 2 & 2 & 7 \\ 2 & 4 & 8 \\ 7 & 8 & 25 \end{pmatrix}
\end{eqnarray}

Si la matriz no es cuadrada siempre aparecera algunos 0s en los autovalres (la diferencia entre las dimensiones de las matrices).
O sea que $\lambda_3^2 = 0$.

Hallemos los $\sigma$. Estos son las raices cuadradas de los $\lambda_i^2$. Es decir,

\begin{eqnarray}
\sigma_1 = \sqrt{\lambda_1^2} = \sqrt{29.62} \approx 5.44  \\
\sigma_2 = \sqrt{\lambda_2^2} = \sqrt{1.38} \approx 1.17 \\
\sigma_3 = 0
\end{eqnarray}

Faltan los $v_i$.




\begin{eqnarray}
v_1 =
\begin{pmatrix}  0.25 \\ 0.31 \\ 0.92 \end{pmatrix} \quad , \quad
v_2 =  
\begin{pmatrix}  0.24 \\ -0.94 \\ 0.24 \end{pmatrix} \quad , \quad
v_3 =  
\begin{pmatrix} {-0.94 \\ -0.16 \\ 0.31}  \end{pmatrix}
\end{eqnarray}

La construccion es la siguiente
\begin{eqnarray}
U =
\begin{pmatrix}
0.55 & -0.83 \\ 0.83 & 0.55
\end{pmatrix}
\end{eqnarray}

\begin{eqnarray}
\Sigma =
\begin{pmatrix}
5.44 & 0 & 0 \\
0 &  1.17  & 0
\end{pmatrix}
\end{eqnarray}

\begin{eqnarray}
V =
\begin{pmatrix}
0.25 & -0.24 & -0.94 \\
0.31 & -0.94 & -0.16 \\
0.92 & -0.34 & 0.31 \\
\end{pmatrix}
\end{eqnarray}

Encontramos que

\begin{eqnarray}
A =
\begin{pmatrix}
0.55 & -0.83 \\ 0.83 & 0.55
\end{pmatrix}
\begin{pmatrix}
5.44 & 0 & 0 \\
0 &  1.17  & 0
\end{pmatrix}
\begin{pmatrix}
0.25 & 0.31  & 0.92 \\
-0.24 & -0.94 & -0.24 \\
-0.94 & -0.16 & 0.31 \\
\end{pmatrix}
\end{eqnarray}




